samples/bootloader/README.rst

```diff
@@ -33,8 +33,10 @@ This is accomplished by the following steps:
      This metadata contains the full public key corresponding to the private key that was used to sign the firmware.
      The bootloader sample checks the public key against a set of provisioned keys.
      Note that to save space, only the hashes of the provisioned keys are stored, and only the hashes of the keys are compared.
-     If the public key in the metadata matches one of the provisioned public key hashes, the image is considered valid.
-     If the public key does not match any of the provisioned hashes, validation fails.
+     If the public key in the metadata matches one of the provisioned public key hashes that are still valid, the image is considered valid.
+     All public key hashes at lower indices than the matching hash are permanently invalidated at this point, meaning images can no longer be validated with those public keys.
+     E.g. if an image is successfully validated with the public key at index 2, public keys 0 and 1 are invalidated.
+     If the public key does not match any of the still valid provisioned hashes, validation fails.
```

Comment:
Should we point out here that this is a way of invalidating a broken key?

Reply:
Good idea. Added a sentence.